src/test/scala/uk/gov/nationalarchives/MainTest.scala

```diff
@@ -282,4 +486,99 @@ class MainTest extends AnyFlatSpec with MockitoSugar with EitherValues {
+        mockPreservicaClient(ioId, coId, metadataElems = Seq(metadata), bitstreamInfo1 = List(bitStreamInfo))
+      val expectedMetadataFileDestinationFilePath = s"$ioId/Preservation_1/$coId/CO_Metadata.xml"
+
+      addToRepo(ioId, repo, data, s"$ioId/name", s"$ioId/Preservation_1/$coId/original/g1/name")
+
+      latestVersion(repo, ioId) must equal(1)
+      runDisasterRecovery(sqsClient, preservicaClient, repo)
+
+      latestVersion(repo, ioId) must equal(2)
+      repo.containsObject(ioId.toString) must be(true)
+      repo.getObject(ioId.toHeadVersion).containsFile(expectedMetadataFileDestinationFilePath) must be(true)
+
+      val metadataStoragePath =
+        repo.getObject(ioId.toHeadVersion).getFile(expectedMetadataFileDestinationFilePath).getStorageRelativePath
+      val metadataContent = Files.readAllBytes(Paths.get(repoDir.toString, metadataStoragePath)).map(_.toChar).mkString
+
+      metadataContent must equal(
+        <AllMetadata>
+      <Test></Test>
+    </AllMetadata>.toString
+      )
+    }
+
+  "runDisasterRecovery" should "write a new version and new CO metadata object if there is a CO update with different metadata" in {
+    val data = "Test"
+    val metadata = <AllMetadata>
+      <Test></Test>
+    </AllMetadata>
+    val ioId = UUID.randomUUID()
+    val coId = UUID.randomUUID()
+    val checksum = DigestUtils.sha256Hex(data)
+    val sqsClient = mockSqs(ContentObjectMessage(coId, s"co:$coId") :: Nil)
+    val bitStreamInfo = BitStreamInfo("name", 1, "", Fixity("SHA256", checksum), 1, Original, None, Some(ioId))
+    val preservicaClient =
+      mockPreservicaClient(
+        ioId,
+        coId,
+        metadataElems = Seq(<DifferentMetadata></DifferentMetadata>),
+        bitstreamInfo1 = Seq(bitStreamInfo)
+      )
+
+    val repoDir = Files.createTempDirectory("repo")
+    val repo = createTestRepo(repoDir)
+    val expectedMetadataFileDestinationFilePath = s"$ioId/Preservation_1/$coId/CO_Metadata.xml"
+    addToRepo(ioId, repo, data, s"$ioId/name", s"$ioId/Preservation_1/$coId/original/g1/name")
+    createExistingMetadataEntry(ioId, ioType.toUpperCase, repo, metadata, expectedMetadataFileDestinationFilePath)
+
+    latestVersion(repo, ioId) must equal(2)
+
+    runDisasterRecovery(sqsClient, preservicaClient, repo)
+
+    latestVersion(repo, ioId) must equal(3)
+
+    val metadataStoragePath =
+      repo.getObject(ioId.toHeadVersion).getFile(expectedMetadataFileDestinationFilePath).getStorageRelativePath
+    val metadataContent =
+      trim(XML.loadString(Files.readAllBytes(Paths.get(repoDir.toString, metadataStoragePath)).map(_.toChar).mkString))
+
+    metadataContent must equal(
+      trim(<AllMetadata>
+        <DifferentMetadata></DifferentMetadata>
+      </AllMetadata>)
+    )
+  }
+
+  "runDisasterRecovery" should "throw an error if the OCFL repository returns an unexpected error" in {
+    val id = UUID.randomUUID()
+    val repoDir = Files.createTempDirectory("repo")
```

Comment:
This variable and `expectedMetadataFileDestinationFilePath` aren't being used in this test.